### PR TITLE
PRMDR-362 virus Scanner Infrastructure

### PR DESCRIPTION
**Pull request by AlexHerbertNHS**
Added infra code for Virus Scanner
Reviews welcome as I write the Readme

### CODE REVIEW

File: virusscanner/terraform/main.tf [hunk 1]
```diff
@@ -0,0 +1,96 @@
+terraform {
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = "4.52.0"
```
Comment (NogaNHS) [inline virusscanner/terraform/main.tf:5]: Can we change this to `version = ">= 4.0"`

File: virusscanner/terraform/main.tf [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+resource "aws_subnet" "virus_scanning_subnet2" {
+  availability_zone = "eu-west-2b"
+  vpc_id            = aws_vpc.virus_scanning_vpc.id
+  cidr_block        = "10.0.2.0/24"
+
+  tags = {
+    Name = "Virus scanning subnet for eu-west-2b"
+  }
+}
```
Comment (NogaNHS) [inline virusscanner/terraform/main.tf:40]: Is it possible to use the vpc module here? Also, Should we re use the ARF virus scanner VPC for now to save on resources?

File: virusscanner/terraform/notifications.tf [hunk 3]
```diff
@@ -0,0 +1,24 @@
+data "aws_cloudformation_export" "proactive_notifications_sns_topic" {
+  name = "${aws_cloudformation_stack.s3_virus_scanning_stack.name}-proactive-notifications-sns-topic"
+}
+
+resource "aws_ssm_parameter" "virus_scan_notifications_sns_topic_arn" {
+  name  = "/prs/${var.environment}/virus-scan-notifications-sns-topic-arn"
+  type  = "String"
+  value = data.aws_cloudformation_export.proactive_notifications_sns_topic.value
+}
+
```
Comment (NogaNHS) [inline virusscanner/terraform/notifications.tf:10]: Same, ssm module here?

File: virusscanner/terraform/main.tf (hunk 2 above)
Comment (AlexHerbertNHS) [reply virusscanner/terraform/main.tf:40]: I don't think so, as this is kept in it's own Terraform folder and does not have access to the tfvars files in the general infrastructure terraform folder

Additionally, the ARF virus scanner is no longer in use; anything uploaded to the ARF project goes through the fake virus scanner only and is automatically marked as Clean

File: virusscanner/terraform/main.tf (hunk 1 above)
Comment (AlexHerbertNHS) [reply virusscanner/terraform/main.tf:5]: Sure can!

File: virusscanner/terraform/main.tf (hunk 2 above)
Comment (NogaNHS) [reply virusscanner/terraform/main.tf:40]: The virus scanner is not in use but the vpc is still up.
Yes you're right, I didn't realise this is a new folder outside of the main one.

File: virusscanner/terraform/main.tf (hunk 1 above)
Comment (AlexHerbertNHS) [reply virusscanner/terraform/main.tf]: Done

File: virusscanner/terraform/main.tf [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+
+resource "aws_subnet" "virus_scanning_subnet1" {
+  availability_zone = "eu-west-2a"
+  vpc_id            = aws_vpc.virus_scanning_vpc.id
+  cidr_block        = "10.0.1.0/24"
+
+  tags = {
+    Name = "Virus scanning subnet for eu-west-2a"
+  }
```
Comment (NogaNHS) [inline virusscanner/terraform/main.tf:29]: Please add extra tags for consistency with the other terraform folder?

**Review by SRAlexander — CHANGES_REQUESTED** [review]

File: virusscanner/terraform/main.tf (hunk 4 above)
Comment (AlexHerbertNHS) [reply virusscanner/terraform/main.tf:29]: Done

File: virusscanner/README.md [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+This module is for the Virus Scanning subsystem only
+More details can be found in Confluence [here](https://gpitbjss.atlassian.net/wiki/spaces/TW/pages/12406227067/Virus+Scanner+maintenance)
+
+## Deployment
+
+Navigate to the Terraform folder
+`./virusscanner/terraform`
+
+awsume into your role 
+`awsume nhse-dev`
```
[the remark this answers is not in the document]
Comment (AlexHerbertNHS) [reply virusscanner/README.md:12]: Done

File: virusscanner/README.md [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+DynamoDB table:
+
+`virus-scanner-terraform-lock`
+
+## Configuration
+
+The virus scanner will inhabit the same VPC as other resources for the project.
+You will need to add the ID of the VPC you want to use to the relevant `tfvars` file
+
+`vpc_id      = "vpc-0123456789abcdef"`
```
Comment (SRAlexander) [inline virusscanner/README.md:27]: Awesome

**Review by SRAlexander — APPROVED** [review]

**Pull request merged**